crates/ruff/src/rules/flake8_tidy_imports/relative_imports.rs

```diff
@@ -99,19 +99,29 @@ fn fix_banned_relative_import(
             parts.pop();
         }
 
-        let call_path = from_relative_import(&parts, module.unwrap());
-        let module_name = call_path.as_slice();
+        let module_name = if let Some(module) = module {
+            let call_path = from_relative_import(&parts, module);
+            call_path.as_slice().join(".")
+        } else {
+            if parts.len() > 1 {
+                let module = parts.last().unwrap();
+                let call_path = from_relative_import(&parts, module);
+                call_path.as_slice().join(".")
+            } else {
```

Comment:
The best I could come up with was to do the validation as part of this scope:

```rust
let module_name = if let Some(module) = module.or_else(|| {
    if parts.len() > 1 {
        Some(parts.last().unwrap())
    } else {
        None
    }
}) {
    let call_path = from_relative_import(&parts, module);
    let parts = call_path.as_slice();

    // Require import to be a valid PEP 8 module:
    // https://python.org/dev/peps/pep-0008/#package-and-module-names
    if parts.iter().any(|f| !is_lower_with_underscore(f)) {
        return None;
    }

    parts.join(".")
} else {
    // Require import to be a valid PEP 8 module:
    // https://python.org/dev/peps/pep-0008/#package-and-module-names
    if parts.iter().any(|f| !is_lower_with_underscore(f)) {
        return None;
    }

    parts.join(".")
};
```

But, I'm not convinced it's any better.

Reply:
I ended up doing something like this, but let me know if you disagree with the current version. Just merging for expediency.